contracts/mirror_factory/src/contract.rs

```diff
@@ -254,16 +258,13 @@ pub fn whitelist(
         return Err(StdError::generic_err("unauthorized"));
     }
 
-    if read_params(deps.storage).is_ok() {
```

Comment:
if the token instantiation fails, we would have params stored, so we would not be able to whitelist again. Its better to remoe this check